### PR TITLE
refactor: updating the authentication to use `go-azure-sdk`

### LINKED ISSUE #72: auth: refactoring to use `hashicorp/go-azure-sdk`
The auth library that this SDK relies on (within `github.com/hashicorp/go-azure-helpers/authentication`) has been replaced by the new `auth` and `environments` packages within `github.com/hashicorp/go-azure-sdk/sdk`) - as such we should look to update the existing Authorizers and Environments to take advantage of this.

So that we can break this refactoring up by Service, it'd be worth doing this in a backwards compatible way (that is, retrieving both the Environment from both the `autorest` and `environments` packages - which will allow #68 to be done gradually.

### PR DESCRIPTION
**Pull request by tombuildsstuff**
This PR fixes #72 - switching the authentication layer to use `hashicorp/go-azure-sdk` (and it's autorest wrappers) rather than using `hashicorp/go-azure-sdk`.

This will enable #68 by allowing the new auth layer to be used to build new SDK Clients, both to allow using the SDK Clients from `hashicorp/go-azure-sdk` (for creating the Resource Group/Storage Account) and within this SDK (so we can switch these over from using `Azure/go-autorest`'s base layer to `hashicorp/go-azure-sdk`).

FWIW - I've intentionally left a TODO in here for adding logging support, since that's tracked in #75.

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
Tests pass:

```
 $ ACCTEST=1 envchain azurerm go test -v ./storage/2020-08-04/blob/accounts -timeout=300m
=== RUN   TestContainerLifecycle
    set_service_properties_test.go:67: [DEBUG] Waiting 2 seconds..
--- PASS: TestContainerLifecycle (61.20s)
PASS
ok  	github.com/tombuildsstuff/giovanni/storage/2020-08-04/blob/accounts	61.254s
```

File: storage/internal/testhelpers/client.go [hunk 1]
```diff
@@ -95,119 +100,111 @@ func (client Client) buildTestResources(ctx context.Context, resourceGroup, name
...
 	}
-
-	return apiClient, nil
-}
-
-func buildAPIClient(config *authentication.Config, env azure.Environment) (*Client, error) {
-	oauthConfig, err := config.BuildOAuthConfig(env.ActiveDirectoryEndpoint)
-	if err != nil {
-		return nil, err
+	if env == nil {
```
Comment (jackofallops) [inline storage/internal/testhelpers/client.go:140]: Should this be?
```suggestion
	if autorestEnv == nil {
```

**Review by jackofallops — APPROVED** [review]
THanks @tombuildsstuff - one comment below, otherwise LGTM 👍

File: storage/internal/testhelpers/client.go (hunk 1 above)
Comment (tombuildsstuff) [reply storage/internal/testhelpers/client.go]: indeed, copy/paste fail - thanks

**Pull request merged**